File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // Naive approach is to use hashing. 
        // unordered_map<int, int> mpp; 
        // for(int i = 0; i < nums.size(); i++)mpp[nums[i]]++; 
        // for(const auto &pair: mpp){
        //     if (pair.second > nums.size()/2)return pair.first; 
        // }
        // return -1; // Place holder.
        int e = 0, cnt = 0; 
        for(int i = 0; i < nums.size(); i++){
            if (cnt == 0)e = nums[i]; 
            cnt += (nums[i] == e)? 1: -1; 
        }
        return e; 
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        # Naive approach. 
        zeroCnt, product = 0, 1
        for num in nums: 
            if (num): 
                product *= num 
            else: 
                zeroCnt += 1
        if (zeroCnt  > 1):
            return [0] * len(nums)
        product_array = [0] * len(nums)
        for i, num in enumerate(nums): 
            if (zeroCnt): product_array[i] = 0 if num else product 
            else: product_array[i] = product // num
        return product_array
        